### PR TITLE
ErrNoSignaturesFound should be used when there is no signature attached to an image.

### PR DESCRIPTION
**Pull request by zhaoyonghe**
#### Summary

This PR closes https://github.com/sigstore/cosign/issues/3525.

#### Release Note

NONE

#### Documentation

No documentation change.

### CODE REVIEW

File: pkg/cosign/verify.go [hunk 1]
```diff
@@ -592,7 +592,7 @@ func verifySignatures(ctx context.Context, sigs oci.Signatures, h v1.Hash, co *C
 	}
 
 	if len(sl) == 0 {
-		return nil, false, &ErrNoMatchingSignatures{
+		return nil, false, &ErrNoSignaturesFound{
```
Comment (haydentherapper) [inline pkg/cosign/verify.go:595]: Update error message?

**Review by haydentherapper — COMMENTED** [review]
Any test updates?

File: pkg/cosign/verify.go (hunk 1 above)
Comment (zhaoyonghe) [reply pkg/cosign/verify.go:595]: updated

**Comment by zhaoyonghe** [comment]
> Any test updates?

The current tests only check whether an error occurs or not. I do not see them checking the return error type and message. Therefore, this change should not break any current tests.

**Comment by haydentherapper** [comment]
What about adding a test that covers this code  path?

**Comment by zhaoyonghe** [comment]
> What about adding a test that covers this code path?

Yes. I can do that. Do you mean checking the [exit code](https://github.com/sigstore/cosign/blob/v2.2.3/cmd/cosign/errors/exit_codes.go) in the [test scripts](https://github.com/sigstore/cosign/tree/v2.2.3/test)? 

Now we indeed return `NoMatchingSignature` (12) in `ImageWithoutSignature` (10) cases.

**Comment by haydentherapper** [comment]
It can be simpler - Add a unit test that calls `verifySignatures` with signature struct without signatures and verifies the returned error.

**Review by haydentherapper — APPROVED** [review]
Thanks!

**Pull request merged**